### PR TITLE
/screenshot api: scrollPage to trigger lazyloaded elements

### PR DESCRIPTION
**Pull request by tomasc**
implements #1922

needs to be tested locally before merging

### CODE REVIEW

File: functions/screenshot.js [hunk 1]
```diff
@@ -18,6 +18,36 @@
  * @param args.page - object - Puppeteer's page object (from await browser.newPage)
  * @param args.context - object - An object of parameters that the function is called with. See src/schemas.ts
  */
+const scrollThroughPage = async (page) => {
+  // Scroll to page end to trigger lazy loading elements
+  await page.evaluate(() => {
+    const scrollInterval = 100;
+    const scrollStep = Math.floor(window.innerHeight / 2);
+    const bottomThreshold = 400;
```
Comment (joelgriffith) [inline functions/screenshot.js:26]: Could probably be derived from https://pptr.dev/#?product=Puppeteer&version=v11.0.0&show=api-pageviewport ?

```js
const { height: bottomThreshold } = await page.viewport();
```

File: functions/screenshot.js [hunk 2]
```diff
@@ -18,6 +18,36 @@
...
+        if (document.body.scrollHeight - bottomPos() < bottomThreshold) {
+          window.scrollTo(0, 0);
+          setTimeout(resolve, 500);
+          return;
+        }
+
+        setTimeout(scrollDown, scrollInterval);
+      }
+
+      setTimeout(reject, 30000);
```
Comment (joelgriffith) [inline functions/screenshot.js:45]: Might not need to reject since the function will timeout on its own in cases of infinite scroll pages.

**Comment by andyMrtnzP** [comment]
I've tested the feature as it is now. Take for instance Discord's webpage, which would regularly be screenshootted as this:
FpAZ5z59o(1)

Using `"scrollPage": true` does render it completely:
FpAZ5z59o

It's a good idea to implement the changes before merging, though
`

**Comment by tomasc** [comment]
@andyMrtnzP thanks.
As for the changes, I think `window.innerHeight` might still be fine, since `viewport` is not available under `page.evaluate()`.
I will look into removing the timeout.

**Comment by tomasc** [comment]
@joelgriffith see my comment above. Id just merge this as-is.

**Comment by tomasc** [comment]
@joelgriffith how about merging this? It is very useful and would be great to have. Thank you!

File: functions/screenshot.js (hunk 1 above)
Comment (tomasc) [reply functions/screenshot.js]: Implemented

**Comment by tomasc** [comment]
@joelgriffith any chance for this to be merged please?

**Review by joelgriffith — APPROVED** [review]

**Comment by joelgriffith** [comment]
Better a few months (quarters?) late then never is what I always say. And I'd know cause I'm the sayer of my sayings.

**Comment by tomasc** [comment]
Won der ful!
Thank you @joelgriffith!

**Pull request merged**